src/test/java/spoon/test/trycatch/TryCatchTest.java

```diff
@@ -442,4 +444,60 @@ public void testTryWithVariableAsResource() {
 		tryStmt.removeResource(ctResource);
 
 	}
+
+	@Nested
+	class AddCatcherAt {
+		@Test
+		void addsCatcherAtTheSpecifiedPosition() {
+			// contract: the catcher should be added at the specified position
+			// arrange
+			Factory factory = createFactory();
+
+			CtTry tryStatement = factory.createTry();
+
+			CtCatch first = factory.createCatch();
+			first.setParameter(factory
+					.createCatchVariable()
+					.setType(factory.Type().createReference(IOException.class)));
+
+			CtCatch second = factory.createCatch();
+			second.setParameter(factory
+					.createCatchVariable()
+					.setMultiTypes(List.of(
+							factory.Type().createReference(ExceptionInInitializerError.class),
+							factory.Type().createReference(NoSuchFieldException.class))));
+
+			CtCatch third = factory.createCatch();
+			third.setParameter(factory
+					.createCatchVariable()
+					.setType(factory.Type().createReference(NoSuchMethodException.class)));
+
+			// act
+			tryStatement.addCatcherAt(0, third);
+			tryStatement.addCatcherAt(0, first);
+			tryStatement.addCatcherAt(1, second);
+
+			// assert
+			assertThat(tryStatement.getCatchers(), contains(first, second, third));
+		}
+
+		@Test
+		void throwsOutOfBoundsException_whenPositionIsOutOfBounds() {
+			// contract: `addCatcherAt` should throw an out-of-bounds exception when the specified position is out of
+			// bounds of the catcher collection
```

Comment:
That's not exactly correct, as `collection.size()` *is* out of bounds, but only `collection.size() + 1` and above (and < 0) throws an exception. More important than the contract here, this should probably also documented in the javadoc of the new method. Maybe we can just say that the behavior corresponds to `List#add(int, E)`?

Reply:
> Maybe we can just say that the behavior corresponds to List#add(int, E)

I think this should be sufficient. They explicitly say [here](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/List.html#add(int,E)) when `IndexOutOfBoundsException` is thrown.

Reply:
Yep, that's good I guess.